server/questions.js

```diff
@@ -544,4 +544,62 @@ exports.voteReply = function (replyId, vote, userId, callback) {
             return callback('Could not submit the vote on the reply', null);
         }
     });
+}
+
+exports.isUserLocked = function(userId, question, callback){
+    var lastSubmissionTime;
+    var currentDate = common.getDateObject();
+
+    // check if user already got the question correct
+    for (var obj = 0; obj < question.correctAttempts.length; obj ++){
+        if(question.correctAttempts[obj]['userId'] == userId){
+            return callback(false,null);
+        }
+    }
+    // find the lastSubmissionTime for this user
+    for (var obj = 0; obj < question.userSubmissionTime.length; obj++){
```

Comment:
Seems inefficient to go through this whole object every time, do you think there could be a better way?

Reply:
right, lets see

Reply:
i can add break;